docs/designs/0001-golang-1-21.md

```diff
@@ -0,0 +1,206 @@
+
+And now the "Crème de la crème" of Go toolchain downloads and verification. Remember the section intro about Go seemingly never looking into `GOMODCACHE` for toolchains? It actually does, but guess what, it needs to verify the toolchains checksum (see [Use toolchain with an offline cache](#use-toolchain-with-an-offline-cache). 
+
+### Example invocations
+This section serves just as a reference to various invocations of Go forcing the toolchain usage.
+
+#### Fetching toolchain with checksum disabled
+```
+$ GOTOOLCHAIN=auto go mod download
+go: downloading go1.21.2 (linux/amd64)
+go: download go1.21.2: golang.org/toolchain@v0.0.1-go1.21.2.linux-amd64: verifying module: checksum database disabled by GOSUMDB=off
+
+# try again to showcase Go will download the toolchain again (but let it pass the verification)
+$ GOTOOLCHAIN=auto GOSUMDB=sum.golang.org go mod download
+go: downloading go1.21.2 (linux/amd64)
+...
+go: downloading golang.org/x/text v0.0.0-20170915032832-14c0d48ead0c
+# get https://proxy.golang.org/golang.org/x/text/@v/v0.0.0-20170915032832-14c0d48ead0c.zip
+# get https://proxy.golang.org/golang.org/x/text/@v/v0.0.0-20170915032832-14c0d48ead0c.zip: 200 OK (0.007s)
+# get https://proxy.golang.org/golang.org/x/text/@v/v0.0.0-20170915032832-14c0d48ead0c.mod
+# get https://proxy.golang.org/golang.org/x/text/@v/v0.0.0-20170915032832-14c0d48ead0c.mod: 200 OK (0.005s)
+...
+```
+
+#### Use toolchain with an offline cache
+```
+$ GOTOOLCHAIN=auto GOSUMDB=off GOMODCACHE=/tmp/gomod/ go mod download --json
+go: golang.org/toolchain@v0.0.1-go1.21.2.linux-amd64: verifying module: checksum database disabled by GOSUMDB=off
+```
+
+#### Fetching with GOMODCACHE and GOPROXY
+```
+$ GOTOOLCHAIN=auto GOSUMDB=off GOMODCACHE=/tmp/gomod/ GOPROXY=file:///tmp/gomod/cache/download go mod download --json
+    {
+	"Path": "golang.org/x/text",
+	"Version": "v0.0.0-20170915032832-14c0d48ead0c",
+	"Info": "/tmp/gomod/cache/download/golang.org/x/text/@v/v0.0.0-20170915032832-14c0d48ead0c.info",
+	"GoMod": "/tmp/gomod/cache/download/golang.org/x/text/@v/v0.0.0-20170915032832-14c0d48ead0c.mod",
+	"Zip": "/tmp/gomod/cache/download/golang.org/x/text/@v/v0.0.0-20170915032832-14c0d48ead0c.zip",
+	"Dir": "/tmp/gomod/golang.org/x/text@v0.0.0-20170915032832-14c0d48ead0c",
+	"Sum": "h1:qgOY6WgZOaTkIIMiVjBQcw93ERBE4m30iBm00nkL0i8=",
+	"GoModSum": "h1:NqM8EUOU14njkJ3fqMW+pc6Ldnwhi/IjpwHt7yyuwOQ="
+}
+...
+```
+
+## Proposed solutions
```

Comment:
I think some standardization is needed also for ^this H2 heading, but I agree the rest of H3+ headings can be left free form and not limit the writer.